library_generation/owlbot/synthtool/gcp/common.py

```diff
@@ -0,0 +1,175 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import os
+import re
+import sys
+import shutil
+import yaml
+from pathlib import Path
+from typing import Dict, List, Optional
+import jinja2
+import logging
+
+from synthtool import _tracked_paths
+from synthtool.sources import templates
+
+logger = logging.getLogger()
+logger.setLevel(logging.DEBUG)
+
+
+DEFAULT_TEMPLATES_PATH = "synthtool/gcp/templates"
+LOCAL_TEMPLATES: Optional[str] = os.environ.get("SYNTHTOOL_TEMPLATES")
+
+# Originally brought from gcp/partials.py.
+# These are the default locations to look up
+_DEFAULT_PARTIAL_FILES = [
+    ".readme-partials.yml",
+    ".readme-partials.yaml",
+    ".integration-partials.yaml",
+]
+
+
+class CommonTemplates:
+    def __init__(self, template_path: Optional[Path] = None):
+        if LOCAL_TEMPLATES is None:
+            logger.error("env var SYNTHTOOL_TEMPLATES must be set")
+            sys.exit(1)
+        logger.debug(f"Using local templates at {LOCAL_TEMPLATES}")
+        self._template_root = Path(LOCAL_TEMPLATES)
+        self._templates = templates.Templates(self._template_root)
+        self.excludes = []  # type: List[str]
+
+    def _generic_library(self, directory: str, relative_dir=None, **kwargs) -> Path:
+        # load common repo meta information (metadata that's not language specific).
+        if "metadata" in kwargs:
+            self._load_generic_metadata(kwargs["metadata"], relative_dir=relative_dir)
+            # if no samples were found, don't attempt to render a
+            # samples/README.md.
+            if "samples" not in kwargs["metadata"] or not kwargs["metadata"]["samples"]:
+                self.excludes.append("samples/README.md")
+
+        t = templates.TemplateGroup(self._template_root / directory, self.excludes)
+
+        if "repository" in kwargs["metadata"] and "repo" in kwargs["metadata"]:
+            kwargs["metadata"]["repo"]["default_branch"] = _get_default_branch_name(
```

Comment:
Do you know which template uses this key?

Reply:
No template uses it, removing